### PR TITLE
Aligning deployments screen w.r.t specifications

### PR DESCRIPTION
**Pull request by rakol-ms**
Signed-off-by: Rajendra Kolli <rakol@microsoft.com>

deployment_screen

### CODE REVIEW

**Comment by rakol-ms** [comment]
svc_alignment
deployment_screen

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 1]
```diff
@@ -47,21 +43,44 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
         }
     }
 
-    private static _getDeploymentItems(deploymentList: V1DeploymentList, replicaSetList: V1ReplicaSetList):
-        IDeploymentItem[] {
-        let items: IDeploymentItem[] = [];
+    private _getDeploymentListView() {
+        let renderList:JSX.Element[] =[];
+        DeploymentsComponent._generateDeploymentReplicaSetMap(this.props.deploymentList, this.props.replicaSetList).forEach(entry => {
+           renderList.push( <ListComponent
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: make sure we have margin-top for the second onwards UI you have

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 2]
```diff
@@ -47,21 +43,44 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
...
-    private static _getDeploymentItems(deploymentList: V1DeploymentList, replicaSetList: V1ReplicaSetList):
-        IDeploymentItem[] {
-        let items: IDeploymentItem[] = [];
+    private _getDeploymentListView() {
+        let renderList:JSX.Element[] =[];
+        DeploymentsComponent._generateDeploymentReplicaSetMap(this.props.deploymentList, this.props.replicaSetList).forEach(entry => {
+           renderList.push( <ListComponent
+                className={css("dc-list-content", "depth-16")}
+                headingContent={DeploymentsComponent._getHeadingContent(entry.deployment)}
+                items={DeploymentsComponent._getDeploymentItems(entry.deployment, entry.replicaSets)}
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: do you want to name _getDeploymentItems this as _getDeploymentReplicaSetItems

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 3]
```diff
@@ -47,21 +43,44 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
...
         });
+        return deploymentReplicaSetMap;
+    }
+
+    private static _getDeploymentItems(deployment: V1Deployment, replicaSets: V1ReplicaSet[]):
+        IDeploymentItem[] {
+        let items: IDeploymentItem[] = [];
+        replicaSets.forEach((replicaSet, index) => {
+                items.push(DeploymentsComponent._getDeploymentItem(deployment, replicaSet, index, replicaSets.length));
+            });
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: low prio: format, 4 extra spaces befor })

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 4]
```diff
@@ -85,7 +104,10 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
             pods: DeploymentsComponent._getPodsText(replica.status.availableReplicas, replica.status.replicas),
             statusProps: DeploymentsComponent._getPodsStatusProps(replica.status.availableReplicas, replica.status.replicas),
             showRowBorder: (replicaSetLength === (index + 1)),
-            deployment: deployment
+            deployment: deployment,
+            //todo :: how to display images of all containers in a replica pod
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: change the todo from how to display images of all containers in a replica pod
to
should we show all containers in a replica pod

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 5]
```diff
@@ -211,8 +227,35 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
...
+            case ageKey:
+            return (
+                <Ago date={new Date(deployment.creationTimeStamp)} />
+            );
         }
 
         return ListComponent.renderColumn(textToRender || "", ListComponent.defaultColumnRenderer, colDataClassName);
     }
+
+    private static _generateLabels(labels:{[key:string]:string}): ILabelModel[]{
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: remove this function and use util function

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 6]
```diff
@@ -211,8 +227,35 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
...
+                content: format("{0}={1}",key,labels[key])
+            });
+        });
+        return labelModels;
+    }
+    private static _getHeadingContent( deployment:V1Deployment ): JSX.Element {
+        return (
+            <div>
+                <h3>{deployment.metadata.name}</h3>
+                <LabelGroup labelProps={DeploymentsComponent._generateLabels(deployment.metadata.labels)}
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: add propert
fadeOutOverflow={true}

File: src/WebUI/Components/ListComponent.tsx [hunk 7]
```diff
@@ -65,8 +65,8 @@ export class ListComponent<T> extends BaseComponent<IListComponentProperties<T>>
     public static renderTwoLineColumn(primaryText: string, subText: string, className?:string, primaryTextClassName?: string, secondaryTextClassName?:string ) : React.ReactNode {
         return (
             <div className={css("kube-list-col-data overflow-ellipsis", className)}>
-                <div className={css("kube-list-col-data", primaryTextClassName)}>{primaryText}</div>
-                <div className={css("list-secondary-text", secondaryTextClassName)}>{subText}</div>
+                <div className={css("kube-list-col-data overflow-ellipsis", primaryTextClassName)}>{primaryText}</div>
```
Comment (madhuv-msft) [inline src/WebUI/Components/ListComponent.tsx]: when overflow happens, how user can see the content, tooltip?

File: src/WebUI/Components/ServicesComponent.scss [hunk 8]
```diff
@@ -9,20 +9,16 @@
...
     display: inline-flex;
     align-items: center;
-    font-size: 16px;
 }
 
 .sc-col-primary-text {
     font-weight: $fontWeightSemiBold;
-    font-size: $fontSizeL;
 }
 .sc-col-secondary-text {
```
Comment (madhuv-msft) [inline src/WebUI/Components/ServicesComponent.scss]: same as .sc-col-header , do you want to combine them?

File: src/WebUI/Types.d.ts [hunk 9]
```diff
@@ -3,10 +3,11 @@
     Licensed under the MIT license.
 */
 
-import { V1Deployment, V1DeploymentList, V1PodList, V1ReplicaSetList, V1Service, V1ServiceList } from "@kubernetes/client-node";
+import { V1Deployment, V1DeploymentList, V1PodList, V1ReplicaSetList, V1Service, V1ServiceList, V1ReplicaSet } from "@kubernetes/client-node";
 import { IObservable } from "azure-devops-ui/Core/Observable";
 import { IStatusProps } from "azure-devops-ui/Status";
 import { IBaseProps } from "office-ui-fabric-react/lib/Utilities";
+import { ILabelModel } from "azure-devops-ui/Label";
```
Comment (madhuv-msft) [inline src/WebUI/Types.d.ts]: low prio: you may not be using import

File: src/WebUI/Types.d.ts [hunk 10]
```diff
@@ -25,6 +26,8 @@ export interface IDeploymentItem {
     statusProps?: IStatusProps;
     showRowBorder?: boolean;
     deployment?: V1Deployment;
+    image: string;
```
Comment (madhuv-msft) [inline src/WebUI/Types.d.ts]: is it mandatory to have an image for deployment? if not, make this as image?: string

File: src/WebUI/Components/DeploymentsComponent.scss [hunk 11]
```diff
@@ -10,4 +16,16 @@
     &.first-replica {
         margin-top: 5px;
     }
-}
\ No newline at end of file
+}
+
+.dc-col-content {
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.scss]: if common css can be used across, then we should make them as common

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 12]
```diff
@@ -47,21 +44,48 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
         }
     }
 
-    private static _getDeploymentItems(deploymentList: V1DeploymentList, replicaSetList: V1ReplicaSetList):
-        IDeploymentItem[] {
-        let items: IDeploymentItem[] = [];
+    private _getDeploymentListView() {
+        let renderList:JSX.Element[] =[];
+        DeploymentsComponent._generateDeploymentReplicaSetMap(this.props.deploymentList, this.props.replicaSetList).forEach((entry, index) => {
+            let columnClassName = css("dc-list-content", "depth-16");
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: const columnClassName = css("dc-list-content", "depth-16", index > 0 ? "replica-with-pod-list" : "");

File: src/WebUI/Components/DeploymentsComponent.tsx [hunk 13]
```diff
@@ -211,8 +232,27 @@ export class DeploymentsComponent extends BaseComponent<IDeploymentsComponentPro
...
         }
 
         return ListComponent.renderColumn(textToRender || "", ListComponent.defaultColumnRenderer, colDataClassName);
     }
+
+    private static _getHeadingContent( deployment:V1Deployment ): JSX.Element {
+        return (
+            <div>
+                <h3>{deployment.metadata.name}</h3>
+                <LabelGroup labelProps={Utils.getUILabelModelArray(deployment.metadata.labels)}
```
Comment (madhuv-msft) [inline src/WebUI/Components/DeploymentsComponent.tsx]: <LabelGroup labelProps={Utils.getUILabelModelArray(deployment.metadata.labels)}
                    wrappingBehavior={WrappingBehavior.OneLine}
                    fadeOutOverflow={true}>
                </LabelGroup>

low prio: we should make this as a separate component, jsut a thought

File: src/WebUI/Components/ListComponent.tsx [hunk 14]
```diff
@@ -65,8 +65,16 @@ export class ListComponent<T> extends BaseComponent<IListComponentProperties<T>>
     public static renderTwoLineColumn(primaryText: string, subText: string, className?:string, primaryTextClassName?: string, secondaryTextClassName?:string ) : React.ReactNode {
         return (
             <div className={css("kube-list-col-data overflow-ellipsis", className)}>
-                <div className={css("kube-list-col-data", primaryTextClassName)}>{primaryText}</div>
-                <div className={css("list-secondary-text", secondaryTextClassName)}>{subText}</div>
+                <div className={css("kube-list-col-data overflow-ellipsis", primaryTextClassName)}>
+                    <TooltipHost content={primaryText} overflowMode={TooltipOverflowMode.Parent}>
```
Comment (madhuv-msft) [inline src/WebUI/Components/ListComponent.tsx]: a common component?

File: src/WebUI/Components/ServicesComponent.scss [hunk 15]
```diff
@@ -7,22 +7,14 @@
     justify-content: space-around;
 }
```
Comment (madhuv-msft) [inline src/WebUI/Components/ServicesComponent.scss]: common css if possible, did not check all of the settings

**Comment by rakol-ms** [comment]
deployment_screen
svc_alignment

**Pull request merged**